### DISCUSSION TITLE
Use Regex for Replacing text in a note

### DISCUSSION THREAD

**Discussion by irmod** [damionx7/Zettel-Notes-Documentation#212] [Feature Requests]
Tell me, it looks like I couldn’t find the functionality for replacing text in a note...

For me, replacing text in a note (and global replacement in the Repository) takes the next place after organizing lists

I ask for your help

**Comment by damionx7** [comment]
While in the editor, click on the search button on the top bar and search for some text. After that a new replace button appears on the top bar, click on it to open the text and replace dialog.

![Screenshot_2024-10-30-02-18-53-63_eb23cbbb1590a46a5676245954a0583b.jpg](https://github.com/user-attachments/assets/c83bc5ce-ff70-45e1-be36-1e56950d11ef)

**Comment by irmod** [comment]
@damionx7,  thank you for your attention and quick response!

I am very pleased that the wonderful application "Zettel Notes" has such an author!!!

Please could you add regular expression support to Search and Replace?

Examples for regular expressions:
 - delete two consecutive empty lines;
 - remove extra spaces between the list marker and the text;
 - remove two or more spaces in a paragraph;
 - etc.

Thank you for your great work

I have attached this image just as an example of my proposal:

![Screenshot_2024-10-30-15-22-15-749_com.ekartoyev.enotes.png](https://github.com/user-attachments/assets/1d511bb9-1bf1-4ce7-9a0b-6296215ed861)

**Comment by damionx7** [comment]
I have added this feature. Its still in testing phase, so always backup your notes before using search and replace using regex. (v2.9.3)

**Comment by irmod** [comment]
@damionx7 thank you for your work! I will carefully test the new version of Zettel Notes
